test.js

```diff
@@ -329,6 +329,106 @@ test('register', function (t) {
       })
   })
 
+  t.test('secret as a function returning a promise', function (t) {
+    t.plan(2)
+
+    const fastify = Fastify()
+    fastify.register(jwt, {
+      secret: function (request, token) {
+        return Promise.resolve('some-secret')
+      }
+    })
+
+    fastify.post('/sign', function (request, reply) {
+      reply.jwtSign(request.body)
+        .then(function (token) {
+          return reply.send({ token })
+        })
+    })
+
+    fastify.get('/verify', function (request, reply) {
+      return request.jwtVerify()
+    })
+
+    fastify
+      .ready()
+      .then(function () {
+        fastify.inject({
+          method: 'post',
+          url: '/sign',
+          payload: { foo: 'bar' }
+        }).then(function (signResponse) {
+          const token = JSON.parse(signResponse.payload).token
+          t.ok(token)
+
+          fastify.inject({
+            method: 'get',
+            url: '/verify',
+            headers: {
+              authorization: `Bearer ${token}`
+            }
+          }).then(function (verifyResponse) {
+            const decodedToken = JSON.parse(verifyResponse.payload)
+            t.is(decodedToken.foo, 'bar')
+          }).catch(function (error) {
+            t.error(error)
+          })
+        }).catch(function (error) {
+          t.error(error)
+        })
```

Comment:
use async/await

Reply:
ok this is all existing code just copied over from existing tests to cover the new secret signatures. I'll change it everywhere then